### PR TITLE
[Enhancement] Add option to always use concurrent tablet creation

### PR DESCRIPTION
**Pull request by rcauble**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [X] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [X] I have checked the version labels which the pr will be auto-backported to the target branch
  - [X] 3.1
  - [X] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 1]
```diff
@@ -1639,7 +1639,7 @@ void buildPartitions(Database db, OlapTable table, List<Partition> partitions) t
             numReplicas += partition.getReplicaCount();
         }
 
-        if (partitions.size() >= 3 && numAliveBackends >= 3 && numReplicas >= numAliveBackends * 500) {
+        if (true /*partitions.size() >= 3 && numAliveBackends >= 3 && numReplicas >= numAliveBackends * 500*/) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:1642]: should use a FE config or session variable here.

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 2]
```diff
@@ -1690,7 +1690,7 @@ private void buildPartitionsSequentially(long dbId, OlapTable table, List<Partit
 
     private void buildPartitionsConcurrently(long dbId, OlapTable table, List<Partition> partitions, int numReplicas,
                                              int numBackends) throws DdlException {
-        int timeout = numReplicas / numBackends * Config.tablet_create_timeout_second;
+        int timeout = numBackends * Config.tablet_create_timeout_second;
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:1693]: why we change this logic? i think for a single tablet with multi replicas, we can have multi backends creating replicas for the tablet concurrently, that's why we devide `numBackends` here.

**Comment by nshangyiming** [comment]
the DCO check is failed, plz add signed-off in your commit message

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java (hunk 1 above)
Comment (rcauble) [reply fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java]: Done

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java (hunk 2 above)
Comment (rcauble) [reply fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java]: I have updated it to be conditional based on the flag. If we don't special-case it then for cases when num buckets is extremely low, we end up with a timeout error message like this:

`fail to create tablet: timed out. unfinished replicas(0/0):  timeout=0s`

**Comment by kangkaisen** [comment]
@rcauble Hi， you could refer this doc https://github.com/StarRocks/starrocks/pull/27402/checks?check_run_id=15145722317  to use `git commit -s` fix DCO issue.

**Comment by rcauble** [comment]
> @rcauble Hi， you could refer this doc https://github.com/StarRocks/starrocks/pull/27402/checks?check_run_id=15145722317 to use `git commit -s` fix DCO issue.

I ran these 2 commands. Hope I did it right:

git rebase HEAD~2 --signoff
git push --force-with-lease origin rcauble-patch-1

**Comment by rcauble** [comment]
It seems this is superceeded by this request? So should I revert these changes then?

https://github.com/StarRocks/starrocks/pull/27418

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by sduzh** [comment]
> It seems this is superceeded by this request? So should I revert these changes then?
> 
> #27418

Hi @rcauble, when I submitted my PR [#27418], I was not aware you were also working on this.
Please go ahead and continue with your fix - don't let my PR block you.  
I'm happy to close my PR or make any adjustments needed. Please let me know if you have any other thoughts!

**Comment by rcauble** [comment]
> > It seems this is superceeded by this request? So should I revert these changes then?
> > #27418
> 
> Hi @rcauble, when I submitted my PR [#27418], I was not aware you were also working on this. Please go ahead and continue with your fix - don't let my PR block you. I'm happy to close my PR or make any adjustments needed. Please let me know if you have any other thoughts!

I'm fine reverting -- your PR looks like it is more general. Just one request: can we please add a 3.0 backport for your PR?

**Comment by sduzh** [comment]
> > > It seems this is superceeded by this request? So should I revert these changes then?
> > > #27418
> > 
> > 
> > Hi @rcauble, when I submitted my PR [#27418], I was not aware you were also working on this. Please go ahead and continue with your fix - don't let my PR block you. I'm happy to close my PR or make any adjustments needed. Please let me know if you have any other thoughts!
> 
> I'm fine reverting -- your PR looks like it is more general. Just one request: can we please add a 3.0 backport for your PR?

OK

**Pull request closed without merging**